flexmeasures/data/models/planning/utils.py

```diff
@@ -356,12 +358,17 @@ def get_series_from_quantity_or_sensor(
             event_starts_after=query_window[0],
             event_ends_before=query_window[1],
             resolution=resolution,
+            # resolution=resolution if not boundary_policy else timedelta(0),  # NotImplementedError in timely-beliefs
+            # frequency=resolution,
             beliefs_before=beliefs_before,
             most_recent_beliefs_only=True,
             one_deterministic_belief_per_event=True,
         )
+        if as_instantaneous_events:
+            bdf = bdf.resample_events(timedelta(0), boundary_policy=boundary_policy)
```

Comment:
Nice! I was thinking on the case that bdf was already instantaneous but the method `resample_events` already skips resampling if the BDF resolution already matches that of the sensor.